nntrainer/include/graph.h

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: Apache-2.0-only
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	graph.h
+ * @date	14 August 2020
+ * @brief	This is Graph Class of Neural Network
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#ifndef __GRAPH_H__
+#define __GRAPH_H__
+#ifdef __cplusplus
+
+#include <tensor_op.h>
+#include <vector>
+
+namespace nntrainer {
+
+/**
+ * @class   Graph
+ * @brief   Graph of Tensors and Tensor Operations
+ */
+class Graph {
+public:
+  Graph() {}
+
+  ~Graph() {}
+
+  /**
+   * @brief Register the tensor op in the graph along with its input/output
+   * tensors
+   *
+   * @param op Operation to be registered
+   */
+  void registerTensorOp(TensorOp op) {}
+
+  /**
+   * @brief Get the all the tensors object
+   *
+   * @return std::vector<Tensor>& List of all the tensor objects for the all
+   * registered ops
+   */
+  std::vector<Tensor> &getTensors() { return all_tensors; }
```

Comment:
Q. So, is `std::vector<Tensor>` :cactus: gonna be changed to a map to save inputs and outputs?
I am confused if :cactus: gonna be a storage(for inputs/outputs) or graph or both?

Reply:
In a graph, we have a list of vectors and a list of edges connecting them.
Here, std::vector<Tensor> is the information on each edge.